GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+> the network (e.g., socket hang-up), or a remote service (e.g., a 500 error,
+> failure to connect, or the like).
+
+Notably, operational errors that are not accounted for by the programmer are
+considered unexpected errors.
+
+### Programmer Error
+
+An error that represents a mistake on the programmer's behalf. To paraphrase the
+[joyent error guide][joyent-errors]:
+
+> Programmer errors are bugs in the program. These are things that can always
+> be avoided by changing the code. They can never be handled properly (since by
+> definition the code in question is broken).
+
+By definition, these errors are unexpected.
+
+### Expected versus Unexpected
+
+Errors may be "expected" — anticipated by the program author at time of writing —
+or "unexpected" — unforeseen by the author.
+
+Programmer errors are _always_ unexpected (or else the programmer would not
+have made them!) Operational errors _may be_ unexpected — there may be an
+unaccounted-for operational state that the program enters. Unexpected errors
+are dangerous, and may cause the program to perform undesirable operations. The
+error symposium recommends that programs encountering unexpected errors should
+crash immediately.
+
+∅               | Programmer Error   | Operational Error
+--------------- | ------------------ | ------------------
+**Expected**    | :no_entry_sign:    | :white_check_mark:
+**Unexpected**  | :white_check_mark: | :white_check_mark:
+
+## Post Mortem Analysis
+
+The practice of debugging a program after it has already crashed by inspecting
+the serialized state of the program at the time of error. This usually leverages
+tools like `lldb`, `gdb`, or `mdb` to inspect a "core" (defined below.)
+
+### Abort
+
+The [`abort()`][abort-syscall] system call. Accessible to Node via
+`process.abort()`. Called by V8 when `--abort-on-uncaught-exception` is enabled
+and an exception is thrown without a top-level `catch` handler.
+
+Halts the program and serializes the state of the program (the contents of the
+stack and heap segments, as well as the state of open file descriptors) to a
+"core file." Causes the program to return a non-zero exit status.
+
+### Core / Core dump / Core file
+
+The serialized state of a program, containing the stack and heap segments of
+memory in their entirety. Analyzable via `mdb`, `gdb`, or `lldb`. V8-specific
+plugins are available in the form of `mdb_v8`, `llnode`, and `lldb-v8`, and
+allow users to inspect JavaScript stack frames as well as the frozen object
+graph in memory (including objects pending garbage collection.)
+
+### Stack Frame
+
+A data structure held in the stack segment of memory representing a function
+call. This typically saves the arguments with which the function was called
+as well as a return address to jump to when the call completes.
+
+### Unwinding the Stack / Top of Stack
+
+Stack frames are collected into a stack that grows and shrinks as functions
```

Comment:
Why? If we could keep the _frames_ through asynchronous execution you'd effectively have an asynchronous stack with context and not just stack traces. (It would not get "unwound" at the end, but you'd still get access to it).

This is not a dream, Chrome does this with v8 today (called async stack traces, but keeps track of the parameter values, closure variables and `this` value at each step of the way)